### PR TITLE
refactor(groups.tsx): make groups responsive for small and medium screen

### PR DESCRIPTION
**Pull request by yagopajarino**
re #557

## Description

Refactor the UX code to make the groups stack (My groups and All Credential Groups) responsive for small and medium screens.

Also refactors the search bar and add group button to fit them in small/medium screens

## Does this introduce a breaking change?

-   [ ] Yes
-   [x] No

### CODE REVIEW

**Comment by yagopajarino** [comment]
Hi @vplasencia PR for issue #557. Let me know what you think :)

File: apps/dashboard/src/pages/groups.tsx [hunk 1]
```diff
@@ -186,7 +196,11 @@ export default function GroupsPage(): JSX.Element {
...
                     <Grid
-                        templateColumns="repeat(3, 1fr)"
+                        templateColumns={{
+                            base: "1fr",
+                            md: "repeat(2, 1fr)",
+                            lg: "repeat(3, 1fr)"
+                        }}
                         gap={10}
                         w="100%"
                         mt="60px"
```
Comment (vplasencia) [inline apps/dashboard/src/pages/groups.tsx:206]:
```suggestion
                        templateColumns={{
                            base: "1fr",
                            md: "repeat(2, 1fr)",
                            xl: "repeat(3, 1fr)"
                        }}
                        gap={10}
                        w={{md:"100%"}}
                        mt="60px"
```

**Review by vplasencia — COMMENTED** [review]
Hey @yagopajarino! Thank you very much for this PR. I added some suggestions. 

After the suggestions, it should look like:

https://github.com/user-attachments/assets/39426fe1-a0bf-48b1-bfc5-a4ea32b6310f

It's good to always test how the width changes on smaller screens (to try all possible smaller screens) by using the responsive option in Google Chrome, for example and also try how it would look on different devices with different dimensions.

**Comment by yagopajarino** [comment]
Hi @vplasencia ! Thanks for the feedback. Just made the changes following the suggestions

**Review by vplasencia — APPROVED** [review]
It looks great! Thank you very much! 💯

**Pull request merged**